lib/ex_force.ex

```diff
@@ -160,6 +161,30 @@ defmodule ExForce do
     do:
       do_get_sobject(client, "sobjects/#{sobject_name}/#{field_name}/#{URI.encode(field_value)}")
 
+  @doc """
+  Makes a request to a SObject based on the value of a specified extneral ID field. Works similarly to get_sobject_by_external_id, but accepts methods other than GET.
+
+  See [SObject Rows by External ID](https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/resources_sobject_upsert.htm)
+  Also [Insert or Update (Upsert) a Record Using an External ID](https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/dome_upsert.htm)
+  """
+  @spec request_sobject_by_external_id(client, method, any, field_name, sobject_name, map) ::
```

Comment:
Hm.. so `request_sobject_by_external_id(client, :get, fv, fn, sn, attrs)` would be same with `get_sobject_by_external_id(client, fv, fn, sn, attrs)` - right?

Hm.. I'm slightly more inclined to have separate functions instead of one function taking `request` as an argument - e.g.

```
get_sobject_by_external_id # GET
create_sobject_by_external_id # POST
create_or_update_sobject_by_external_id # PATCH
delete_sobject_by_external_id # DELETE
```

and they share the same function.

Because choosing the HTTP request (`:get`, `:post`, ..) is implementation details - not the high-level behavior.

Any thought?